### ISSUE TITLE
Create modal carousel component

### ISSUE THREAD

**Issue by reemhamz** [mozilla/protocol#856]
### Please describe the use case(s) for this component
There are several pages in Bedrock that use the modal component, which is overlayed with bespoke Javascript to make the modals have carousel-like/slide-like behaviour. They are clickable through directional controls, and will loop when the user reaches the "last" slide in the carousel.

Some examples:
- [Careers pages ](https://www.mozilla.org/en-US/careers/) in the testimonials section
- State of Mozilla 2019-2021 annual reports found in content cards:
   - [2019 annual report](https://www.mozilla.org/en-US/foundation/annualreport/2019/)
   - [2020 annual report](https://www.mozilla.org/en-US/foundation/annualreport/2020/)
   - [2021 annual report](https://www.mozilla.org/en-US/foundation/annualreport/2021/)
- Diversity 2021 reports found in following pages' content cards:
   - [Who we Are page](https://www.mozilla.org/en-US/diversity/2021/who-we-are/)
   - [What we Build page](https://www.mozilla.org/en-US/diversity/2021/what-we-build/)
   - [Beyond our Products page](https://www.mozilla.org/en-US/diversity/2021/beyond-our-products/)

The bespoke modal carousel code was first used in the 2019 SOM annual report, and it's been used in several other pages since.

> **Once is happenstance. Twice is coincidence. Three times is a pattern** 

We should add this modal carousel feature variant to our already-existing modal component in Protocol that we can easily use throughout Bedrock instead of having to always copy-and-paste the same bespoke JavaScript code that was _only_ meant for the 2019 annual report.

### Please provide any designs or previous implementations of the proposed component
- From the Careers pages:
<kbd>![Screen Shot 2023-02-28 at 6 08 30 PM](https://user-images.githubusercontent.com/42309026/221779729-ac9c3a93-c16e-42a6-ad0b-a352e534795e.png)</kbd>

- From the Diversity pages
<kbd>![Screen Shot 2023-02-28 at 6 09 34 PM](https://user-images.githubusercontent.com/42309026/221779957-5d6859c6-b6a1-45a8-b6b8-5d1bf5e12ec2.png)
</kbd>

A worry that I have is accessibility issues that may come with building such a component. It's not a very well-loved component in the web development community because of lack of a11y best practices, disruptiveness to SEO, and can impact performance.
Things to consider are also how disruptive implementing this component may be to already-existing modal carousels in Bedrock, and how what path we'd be taking to structure our modal components in arrays to work as a carousel.
This also requires a great deal of user testing to get right, so it's a triumph of a component if achieved.

**Comment by alexgibson** [comment]
Rather than make a new component for this, would it make sense to add this as a new feature to the existing modal component?

**Comment by reemhamz** [comment]
Oop, that was already the plan -- to extend this feature to the existing modal. I should have mentioned it. Going to add it now, thanks for heads up :)

**Comment by reemhamz** [comment]
Discussed this change in our last Protocol meeting with @craigcook, Emma and Jessica, and we decided that this component shouldn't be developed as it's an accessibility liability and not a very utilized component. 

Going forward, if we receive designs that have modal carousel features, we're going to give the designer + stakeholders other recommendations to make sure we develop more accessible and comfortable experiences to the user.

With that said, we concluded that developing an inline carousel component would be a good middle ground. In this component, we wouldn't be placing any important information that is crucial to the webpage, but rather place secondary information that is just a nice add-on -- a good example of what would be updated to use the inline component would be the Quotes section in the [Careers page](https://www.mozilla.org/en-US/careers/).
